### PR TITLE
Blik (Await) Don't call /details endpoint when payload is missing

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
Don't make "final state" calls (leading to a call to `onAdditionalDetails`) when `payload` is missing. 
Instead show an error state in the component.

## Tested scenarios
Mocked responses with `resultCode:"authorised"` and an empty &/or missing 'payload' to ensure an error is shown and no `/details` call is made.
Mocked responses with `resultCode:"pending"` to confirm that the regular polling to the `/status` endpoint continues until the timeout (there were some concerns, expressed in the ticket) that it switched off after three attempts

**Relates to issue**:  DEV-54329

### CODE REVIEW

File: packages/lib/src/components/internal/Await/Await.tsx [hunk 1]
```diff
@@ -67,7 +80,7 @@ function Await(props: AwaitComponentProps) {
             .catch(({ message, ...response }) => ({
                 type: 'network-error',
                 props: {
-                    ...(message && { message: this.props.i18n.get(message) }),
+                    ...(message && { message: props.getI18n(message) }),
```
Comment (m1aw) [inline packages/lib/src/components/internal/Await/Await.tsx:83]: Why was this changed?

Comment (sponglord) [reply packages/lib/src/components/internal/Await/Await.tsx:83]: Because we don't actually use it.
And if we were to use it I thought we should just pass in a function that retrieves a translation rather than the translation "machine" itself (i.e. `i18n`) - which means the Await comp then needs to know about the existence of things like a `get` function

Comment (sponglord) [reply packages/lib/src/components/internal/Await/Await.tsx:83]: My mistake - I didn't think it ever received a `i18n` prop - but I see it retrieves it directly from `CoreContext`.
I'll change it

File: packages/lib/src/components/internal/Await/Await.tsx [hunk 2]
```diff
@@ -127,9 +140,13 @@ function Await(props: AwaitComponentProps) {
...
             };
 
+            // Reset 'loading' to ensure that it is changes to this value that are the main "engine" that drives this useEffect
+            setLoading(true);
+
+            // Create (another) interval to poll for a result
             setStoredTimeout(setTimeout(statusInterval, delay));
         }
-    }, [loading, timePassed, expired, completed]);
+    }, [loading, expired, completed]);
```
Comment (ribeiroguilherme) [inline packages/lib/src/components/internal/Await/Await.tsx:149]: Are you sure we don't need the `timePassed` here? We rely on it inside this useEffect , and its value gets updated often

Comment (sponglord) [reply packages/lib/src/components/internal/Await/Await.tsx:149]: But I'm going to set it back - 'cos using `loading` as the drive has an odd effect on the UI

File: packages/lib/src/components/internal/Await/types.ts [hunk 3]
```diff
@@ -27,4 +27,5 @@ export interface AwaitComponentProps {
     messageText: string;
     awaitText: string;
     ref: any;
+    getI18n?: (key: string, options?) => string; // Should equate to the Language.get method
```
Comment (ribeiroguilherme) [inline packages/lib/src/components/internal/Await/types.ts:30]: Is this needed? I don't see it being used

**Review by ribeiroguilherme — APPROVED** [review]

File: packages/lib/src/components/internal/Await/types.ts (hunk 3 above)
Comment (sponglord) [reply packages/lib/src/components/internal/Await/types.ts]: you're right it got removed in a recent change to this PR
